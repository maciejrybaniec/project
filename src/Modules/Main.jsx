/* @flow */
/**
 * Render static page.
 * @module Modules/Main
 * @copyright (c) 2016-present maciej.rybaniec@gmail.com
 */

import React, { PureComponent } from 'react';

import Navbar from 'Modules/Navigation/Navbar';

type PropsType = {
    children?: React.Element<*>
};

class Main extends PureComponent {
    /**
    * Set properties validation for component.
    */
    props: PropsType
    /**
     * Render component in DOM.
     * @returns {ReactElement}
     */
    render(): React.Element<*> {
        const { children } = this.props
        return (
            <div className="Main">
                <Navbar />
                <div className="Main__content">
                    {children}
                </div>
            </div>
        );
    }
}

 export default Main;
