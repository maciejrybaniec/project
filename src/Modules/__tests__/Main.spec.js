/**
 * Main component specification.
 * @module Modules/__tests__/Main
 * @copyright (c) 2016-present maciej.rybaniec@gmail.com
 */

import React from 'react';
import Renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Main from '../Main';

describe('<Main />', () => {

    it('should render <Main /> component correctly', () => {
        const tree = Renderer.create(<Main />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render application main container', () => {
        const wrapper = mount(<Main />);
        expect(wrapper.find('.Main').length).toEqual(1);
    });

    it('should render <Navbar /> component', () => {
        const wrapper = mount(<Main />);
        expect(wrapper.find('.Navbar').length).toEqual(1);
    });

    it('should render children components', () => {
        const wrapper = mount(
            <Main>
                <div className="Children"></div>
                <div className="Children"></div>
            </Main>
        );
        expect(wrapper.find('.Children').length).toEqual(2);
    });
});
