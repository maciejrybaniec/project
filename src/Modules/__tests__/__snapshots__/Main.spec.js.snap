exports[`<Main /> should render <Main /> component correctly 1`] = `
<div
  className="Main">
  <nav
    className="Navbar">
    Navigation
  </nav>
  <div
    className="Main__content" />
</div>
`;
